Content.Shared/Stacks/StackComponent.cs

```diff
@@ -78,6 +78,14 @@ public sealed partial class StackComponent : Component
         [DataField("layerStates")]
         [ViewVariables(VVAccess.ReadWrite)]
         public List<string> LayerStates = new();
+
+        // Frontier: transforming Amount, MaxCount in speso stacks 
+        /// <summary>
+        /// An optional function to adjust the layers used for a stack's appearance.
+        /// </summary>
+        [DataField("layerFunction")]
+        public StackLayerFunction? LayerFunction;
```

Comment:
hm this seems really smelly to me. It seems like you're storing a function into a component, which isnt serializable and will break any kind of saving/loading or serialization. The benefits of keeping components to only data, and a fully ECS system.